packages/slate/src/transforms/text.ts

```diff
@@ -70,7 +70,15 @@ export const TextTransforms = {
       }
 
       if (!hanging) {
-        at = Editor.unhangRange(editor, at, { voids })
+        const [, end] = Range.edges(at)
+        const endBlock = Editor.above(editor, {
+          at: end,
+          match: n => Editor.isBlock(editor, n),
+        })
+
+        if (endBlock && !Editor.isVoid(editor, endBlock[0])) {
+          at = Editor.unhangRange(editor, at, { voids })
+        }
```

Comment:
I think this might result in another set of bugs that we don't really intend to introduce.  What if the `endBlock` is a `void` but it's not at the end of the document?  Then, the `void` node would be deleted because the range is not "unuhung".  Are there test cases for hanging ranges with voids in the middle of a document?  If not, we should likely add a test case(s).  And, maybe we consider doing something like the below:

```suggestion
        const [, end] = Range.edges(at)
        const endDoc = Editor.end(editor, []);

        if (!Point.equals(end, endDoc)) {
          at = Editor.unhangRange(editor, at, { voids })
        }
```

Need to think about this behavior a bit more.  The above might not be exactly what we want.  We should pair.